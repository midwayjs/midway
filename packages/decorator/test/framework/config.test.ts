import {
  Config,
  getClassMetadata,
  CONFIG_KEY,
  CLASS_KEY_CONSTRUCTOR,
} from '../../src';

class Test {
  constructor(@Config('aaa') aaa: any) {
    // ignore
  }

  @Config()
  hhh: any;

  @Config('bbb')
  bbb: any;
}

describe('/test/framework/config.test.ts', () => {
  it('config decorator should be ok', () => {
    let data = getClassMetadata(CLASS_KEY_CONSTRUCTOR, Test);
    expect(data).toStrictEqual({
      0: {
        key: 'aaa',
        type: CONFIG_KEY,
      },
    });

    data = getClassMetadata(CONFIG_KEY, Test);
    expect(data).toStrictEqual([
      { key: 'hhh', propertyName: 'hhh' },
      { key: 'bbb', propertyName: 'bbb' },
    ]);
  });
});
