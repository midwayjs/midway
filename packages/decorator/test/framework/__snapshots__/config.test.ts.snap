// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/config.test.ts config decorator should be ok 1`] = `
{
  "bbb": {
    "impl": true,
    "key": "config",
    "metadata": {
      "identifier": "bbb",
    },
    "propertyName": "bbb",
  },
  "hhh": {
    "impl": true,
    "key": "config",
    "metadata": {
      "identifier": undefined,
    },
    "propertyName": "hhh",
  },
}
`;
